### PR TITLE
Fix issue#20, bad handshake.

### PR DESCRIPTION
**Pull request by LeeXiaolan**
This issue is caused by old password using, but no diagnosis information gives out.

### CODE REVIEW

**Comment by jskorpan** [comment]
Can't bring in GPL:ed code I'm afraid. Our code is BSD. Sorry

On 14 apr 2013, at 04:04, "LeeXiaolan" <notifications@github.com<mailto:notifications@github.com>> wrote:

This issue is caused by old password using, but no diagnosis information gives out.

---

You can merge this Pull Request by running

  git pull https://github.com/LeeXiaolan/ultramysql master

Or view, comment on, or merge it at:

  https://github.com/esnme/ultramysql/pull/30

Commit Summary
-   Detect old password using when authentication failed and give out instruction to fix it.

File Changes
-   M lib/Connection.cpphttps://github.com/esnme/ultramysql/pull/30/files#diff-0 (6)

Patch Links:
-   https://github.com/esnme/ultramysql/pull/30.patch
-   https://github.com/esnme/ultramysql/pull/30.diff

**Comment by LeeXiaolan** [comment]
Just merge fdb8b31, it is not gpl code.
Commit 3ab5f25 just personal use.

**Comment by jskorpan** [comment]
Ok I’ll take a look at it shortly, thanks!

File: lib/Connection.cpp [hunk 1]
```diff
@@ -521,6 +521,12 @@ bool Connection::connect(const char *_host, int _port, const char *_username, co
     m_dbgMethodProgress --;
     return false;
   }
+  if (result == 0xfe)
+  {
+    setError ("You are using old password, please update it use PASSWORD(), other than OLD_PASSWORD().", 4, UME_OTHER);
```
Comment (jskorpan) [inline lib/Connection.cpp:526]: I think we could improve the clarity of this message.

Also
How can we be sure that result 0xfe really means this and only this? 

//JT

Comment (LeeXiaolan) [inline lib/Connection.cpp:526]: Check out http://dev.mysql.com/doc/internals/en/connection-phase.html#packet-Protocol::AuthSwitchResponse and http://dev.mysql.com/doc/internals/en/connection-phase.htm#packet-Protocol::OldAuthSwitchRequest

Comment (jskorpan) [inline lib/Connection.cpp:526]: Understood, merging it on friday!
Thanks for the contribution.
//JT

On 20 apr 2013, at 03:50, "Xiaolan.Lee" <notifications@github.com<mailto:notifications@github.com>> wrote:

In lib/Connection.cpp:

> @@ -521,6 +521,12 @@ bool Connection::connect(const char *_host, int _port, const char *_username, co
>      m_dbgMethodProgress --;
>      return false;
>    }
> -  if (result == 0xfe)
> -  {
> -    setError ("You are using old password, please update it use PASSWORD(), other than OLD_PASSWORD().", 4, UME_OTHER);

Check out http://dev.mysql.com/doc/internals/en/connection-phase.html#packet-Protocol::AuthSwitchResponse and http://dev.mysql.com/doc/internals/en/connection-phase.htm#packet-Protocol::OldAuthSwitchRequest

**Comment by jskorpan** [comment]
Can we add a test to the test suite to improve the test coverage?

**Pull request merged**